src/ConfigEditor/ConfigEditor.tsx

```diff
@@ -72,6 +86,19 @@ export function ConfigEditor(props: Props) {
     }
   }, [arn]);
 
+  // Description to show in datasource card in list
+  const { onOptionsChange: propsOnOptionsChange } = props;
+  useEffect(() => {
```

Comment:
This should not be done here: it's triggered for every key stroke on clusterIdentifier Input 😞 

option 1: do this on save
option 2: use onBlur on input rather than onChange

Reply:
I agree, there is no exposed hook for "on save" but we can make it on blur. We need to modify the component in the SDK but it should be simple: https://github.com/grafana/grafana-aws-sdk-react/blob/main/src/sql/ConfigEditor/InlineInput.tsx#L28

In fact, if we can expose all the `Input` properties there we will avoid this kind of issues in the future.

If you create a PR against that repo, you can just bump the version there and it will be automatically released so you can use it here.

Reply:
I was wondering if we could instead make `clusterId` a dropdown by exposing another endpoint (same call without filtering on cluster ID)?
It seems more user friendly to select a cluster rather than typing it?

Reply:
that's a good point, it's actually the experience that you get in the AWS console. The only downside of this is that the API permission that is optional now would become mandatory. We could workaround that allowing people to type the the clusterId anyway in the selector but that also requires changes in the SDK :sweat_smile: (since the wrapper around the Select doesn't allow to set the `allowCustomValue` at the moment).

In any case, since it would be a UX improvement and it's trivial to implement now that we have already the endpoint to get a Cluster I would go for it. I would change the current `Cluster` endpoint for a its plural and return the list of all the clusters (so no additional API requests are needed). WDYT?

Reply:
yes, that sounds good, only having the one API request and get the url from the cluster selected in the dropdown 👍 

Re-permissions, are you saying we should also make changes to the SDK to allow custom values for clusterID? Or make the API permission compulsory?

Reply:
since we can avoid the "breaking" change yes, we should modify the SDK to allow that parameter. Basically we would need to modify this [Select](https://github.com/grafana/grafana-aws-sdk-react/blob/main/src/sql/ResourceSelector.tsx#L112) and modify the props of the wrappers [ResourceSelector](https://github.com/grafana/grafana-aws-sdk-react/blob/main/src/sql/ResourceSelector.tsx#L8) and [ConfigSelect](https://github.com/grafana/grafana-aws-sdk-react/blob/main/src/sql/ConfigEditor/ConfigSelect.tsx#L6)